finality-aleph/src/sync/handler/mod.rs

```diff
@@ -2879,4 +2895,53 @@ mod tests {
             }
         }
     }
+
+    #[tokio::test]
+    async fn accepts_chain_extension_branch() {
+        let (mut h1, mut b1, mut n1, genesis) = setup();
+        let (mut h2, _b2, _n2, _genesis) = setup();
+
+        // Make h1 aware of 2 headers.
+        let branch = grow_light_branch(&mut h1, &genesis, 2, 0);
+
+        // Import the corresponding 2 blocks in h1.
+        for h in branch.clone() {
+            let block = MockBlock::new(h.clone(), true);
+            b1.import_block(block, false);
+            match n1.next().await {
+                Ok(BlockImported(header)) => {
+                    h1.block_imported(header)
+                        .expect("block imported should succeed");
+                }
+                _ => panic!("should notify about imported block"),
+            }
+        }
+
+        println!("kurcze 1");
+
+        // h1 gets the ChainExtension request.
+        let state_h2 = h2.state().unwrap();
+        let action = h1.handle_chain_extension_request(state_h2).unwrap();
+
+        // The response should contian the two blocks.
+        use SimplifiedItem::B;
+        let items = match action {
+            Action::Response(items) => {
+                assert_eq!(
+                    SimplifiedItem::from_response_items(items.clone()),
+                    vec![B(1), B(2)],
+                );
+                items
+            }
+            _ => panic!("should be response"),
+        };
+
+        println!("kurcze 2");
+
+        // The result should be handled correctly, blocks should get imported.
```

Comment:
Wasn't there some way to check whether the import calls happen? Just for completeness. Ignore if it would be annoying, the handler returning no errors should be indication enough in practice.

Reply:
Good point, done